protocol/initproducerid/initproducerid.go

```diff
@@ -0,0 +1,29 @@
+package initproducerid
+
+import "github.com/segmentio/kafka-go/protocol"
+
+func init() {
+	protocol.Register(&Request{}, &Response{})
+}
+
+type Request struct {
+	TransactionalID      string `kafka:"min=v0,max=v4,nullable"`
+	TransactionTimeoutMs int32  `kafka:"min=v0,max=v4"`
+	ProducerID           int64  `kafka:"min=v3,max=v4"`
+	ProducerEpoch        int16  `kafka:"min=v3,max=v4"`
+	// We need at least one tagged field to indicate that this is a "flexible" message
+	// type.
+	_ struct{} `kafka:"min=v2,max=v4,tag"`
+}
+
+func (r *Request) ApiKey() protocol.ApiKey { return protocol.InitProducerId }
+
+type Response struct {
+	ThrottleTimeMs int32    `kafka:"min=v0,max=v4"`
+	ErrorCode      int16    `kafka:"min=v0,max=v4"`
+	ProducerID     int64    `kafka:"min=v0,max=v4"`
+	ProducerEpoch  int16    `kafka:"min=v0,max=v4"`
+	_              struct{} `kafka:"min=v2,max=v4,tag"`
```

Comment:
We generally place these at the top of the struct with a separating whitespace like so: https://github.com/segmentio/kafka-go/blob/fbe3d96cd21174e8d8ef6ce43ab10f2bb21e722a/protocol/createtopics/createtopics.go#L10-L13